### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by mstump**
Kops features and options are poorly documented. I'm proposing that you don't accept features or options unless they're accompanied by the appropriate docs.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ Follow either of the two links above to access the appropriate CLA and instructi
 1. If your proposed change is accepted, and you haven't already done so, sign a Contributor License Agreement (see details above).
 1. If instructed by the repo owners provide a short design document in a PR.
 1. Fork the desired repo, develop and test your code changes. Unit tests are required for most PRs.
+1. All additions of features or options must be documented with example invocations and acceptable input values. If possible the the feature or option must be exposed via the input yaml manifest specified during cluster creation.
```
Comment (KashifSaadat) [inline CONTRIBUTING.md]: Just need to drop the double: `the the feature`

**Comment by justinsb** [comment]
I think the policy we want is that options are documented with descriptive code comments such that they appear in the generated API docs, and that we have a tutorial that allows discovery of each _group_ of options, and then the reference docs are sufficient that someone that has found where to look from the tutorial can easily find and understand the whole family of options (e.g. we don't want to document each etcd option in a tutorial with its own example, but we have a page which says "here's how you select etcd3 and change the heartbeat interval, and see the full options here", and ideally that page is part of a fairly straightforward step-by-step tutorial which users could follow along with / flip-through / search-into)

We certainly have some problems:

* The effort to create a discoverable tutorial stalled out the first time around on PR review, but I'm trying again now.  As always, help with PR review is very welcome!
* We have to make sure that those autogenerated API docs are generated and uploaded somewhere they can actually be referenced by browsers!
* We have to make sure that the code comments are good enough that the reference docs are sufficient
* We have to make sure we have a tutorial that hits on each group
* We'd ideally have good search on all the above

My concern about the "every option must be individually fully documented" policy is that encyclopedic documentation is too large to be discoverable IMO.  We're already approaching that point - it's hard for users to find the relevant documentation, even where that exists.  And that certainly reduces the motivation for documentation.

Are there particular features/options you're thinking of @mstump that need more documentation?  It might be easier to discuss / fix concrete examples here

**Pull request closed without merging**